dali/ft/filecopy.cpp

```diff
@@ -3162,13 +3162,11 @@ void FileSprayer::updateTargetProperties()
                     {
                         Owned<IAttributeIterator> aiter = curSource.properties->getAttributes();
                         //At the moment only clone the topLevelKey indicator (stored in kind), but make it easy to add others.
-                        ForEach(*aiter) {
+                        ForEach(*aiter)
+                        {
                             const char *aname = aiter->queryName();
-                            if (strieq(aname,"@kind")
-                                ) {
-                                if (!curProps.hasProp(aname))
-                                    curProps.setProp(aname,aiter->queryValue());
-                            }
+                            if (!curProps.hasProp(aname))
+                                curProps.setProp(aname,aiter->queryValue());
```

Comment:
@AttilaVamos  Are you sure this won't copy properties when not valid? e.g. if repartions and number of parts in source and destination aren't same, then properties like size , and fileCrc will not be correct.

Reply:
@jakesmith This is the final step of copy(spray, etc) procedure, so all attributes related the target parts like: size, num, crc, etc has been already set. By my best intention this will copy the extras (kind, recordCount, <future_stuff>) only.

Reply:
I'm not sure.
Looking at the code before this, it doesn't necessarily (for example) set compressedSize. It depends on whether 'compressOutput' is set or not.
So if it is not, then it will not be set per part (correct),
but the code you've change here, not will copy the property from the source, if it's present (incorrect).

Similar for fileCrc I think, if calcCRC() is false, the proceeding code will not set it per part, but this copy attribute code will carry it over from the source which would be wrong.

Reply:
Also this block of code is in this conditional block:
```
if (replicate && (distributedSource != distributedTarget) )
```

so only copies attributes if replicate is true?
Meaning still wrong (e.g. missing "kind") if replicate=false ...

Reply:
@AttilaVamos  back to you

Reply:
Yes, you are right, I should rework this.

Reply:
Are the number of parts of target and source always the same at this point?
If they are, then copying recordCount is good, if not, then it will be wrong.

Reply:
That is a good point. No, it isn't (only for key or keydiff files). I will push an update soon.